hail/python/hailtop/hailctl/config/initialize.py

```diff
@@ -0,0 +1,240 @@
+                                       default=True)
+    if create_remote_tmpdir:
+        remote_tmpdir, warnings = await setup_new_remote_tmpdir(
+            bucket_name=maybe_bucket_name,
+            region=region,
+            project=project,
+            username=username,
+            service_account=hail_identity,
+            verbose=verbose
+        )
+    else:
+        remote_tmpdir, warnings = await setup_existing_remote_tmpdir(region, hail_identity, verbose)
+
+    return (remote_tmpdir, region, warnings)
+
+
+async def async_basic_initialize(incremental: bool = False, verbose: bool = False):
+    from hailtop.auth.auth import async_get_userinfo  # pylint: disable=import-outside-toplevel
+    from hailtop.batch_client.aioclient import BatchClient  # pylint: disable=import-outside-toplevel
+    from hailtop.config import DeployConfig  # pylint: disable=import-outside-toplevel
+    from hailtop.config import get_user_config_path  # pylint: disable=import-outside-toplevel
+
+    from ..auth.login import async_login  # pylint: disable=import-outside-toplevel
+    from .cli import set as set_config, list as list_config  # pylint: disable=import-outside-toplevel
+    from .utils import already_logged_into_service, get_default_region  # pylint: disable=import-outside-toplevel
+
+    domain = Prompt.ask('What domain is the Hail service running in?', default='hail.is')
+    namespace = 'default'
+
+    deploy_config = DeployConfig('external', namespace, domain)
+    deploy_config.dump_to_file()
+
+    already_logged_in = await already_logged_into_service()
+    if not already_logged_in:
+        await async_login(namespace)
+
+    user_info = await async_get_userinfo()
+    username = user_info['username']
+    hail_identity = user_info['hail_identity']
+    trial_bp_name = user_info['trial_bp_name']
+
+    batch_client = await BatchClient.create(trial_bp_name, deploy_config=deploy_config)
+
+    async with batch_client:
+        cloud = await batch_client.cloud()
+        supported_regions = await batch_client.supported_regions()
+        default_region = get_default_region(supported_regions, 'gcp')
+
+        if cloud == 'gcp':
+            remote_tmpdir, region, warnings = await initialize_gcp(username, hail_identity, supported_regions, default_region, verbose)
+        else:
+            region = Prompt.ask('Which region should resources be created in?', default=default_region,
+                                choices=supported_regions)
+
+            remote_tmpdir = Prompt.ask(f'Enter a path to an existing remote temporary directory (ex: gs://my-bucket/batch/tmp)')
+            typer.secho(f'WARNING: You will need to grant read/write access to {remote_tmpdir} for account {hail_identity}', fg=typer.colors.MAGENTA)
+            warnings = True
+
+    config_file = get_user_config_path()
+
+    if not incremental:
+        if os.path.isfile(config_file):
+            os.remove(config_file)
```

Comment:
Now that I've gotten this far, I think we should just make a separate command for this. If someone really wants to start fresh they can run `hailctl config clear && hailctl config init`. What do you think?

Reply:
I made a separate command, but I do think `hailctl config init` should always clear any existing environment. Otherwise, there might be parameters set that aren't intended.